File: swift-backend/Sources/App/Models/Account.swift
import Fluent
import Vapor

final class Account: Model, Content {
    static let schema = "accounts"

    @ID(key: .id) var id: UUID?

    @Parent(key: "item_id") var item: PlaidItem

    @Field(key: "plaid_account_id") var plaidAccountId: String

    @Field(key: "name") var name: String

    @Field(key: "mask") var mask: String?

    @OptionalField(key: "official_name") var officialName: String?

    @OptionalField(key: "current_balance") var currentBalance: Decimal?

    @OptionalField(key: "available_balance") var availableBalance: Decimal?

    @OptionalField(key: "iso_currency_code") var isoCurrencyCode: String?

    @OptionalField(key: "unofficial_currency_code") var unofficialCurrencyCode: String?

    @OptionalField(key: "type") var type: String?

    @OptionalField(key: "subtype") var subtype: String?

    @Timestamp(key: "created_at", on: .create) var createdAt: Date?

    @Timestamp(key: "updated_at", on: .update) var updatedAt: Date?

    @Timestamp(key: "deleted_at", on: .delete) var deletedAt: Date?

    init() {}

    init(
        id: UUID? = nil,
        itemId: UUID,
        plaidAccountId: String,
        name: String,
        mask: String?,
        officialName: String?,
        currentBalance: Decimal?,
        availableBalance: Decimal?,
        isoCurrencyCode: String?,
        unofficialCurrencyCode: String?,
        type: String?,
        subtype: String?
    ) {
        self.id = id
        self.$item.id = itemId
        self.plaidAccountId = plaidAccountId
        self.name = name
        self.mask = mask
        self.officialName = officialName
        self.currentBalance = currentBalance
        self.availableBalance = availableBalance
        self.isoCurrencyCode = isoCurrencyCode
        self.unofficialCurrencyCode = unofficialCurrencyCode
        self.type = type
        self.subtype = subtype
    }

}

File: mobile/lib/src/accounts/account.dart
/// A placeholder class that represents an entity or model.
class Account {
  const Account({
    required this.id,
    required this.itemId,
    required this.userId,
    required this.name,
    required this.mask,
    required this.officialName,
    required this.currentBalance,
    required this.availableBalance,
    required this.isoCurrencyCode,
    this.unofficialCurrencyCode,
    required this.type,
    required this.subtype,
    required this.createdAt,
    required this.updatedAt,
    this.deletedAt,
  });

  final String id;
  final String itemId;
  final String userId;
  final String name;
  final String mask;
  final String officialName;
  final double currentBalance;
  final double availableBalance;
  final String isoCurrencyCode;
  final String? unofficialCurrencyCode;
  final String type;
  final String subtype;
  final String createdAt;
  final String updatedAt;
  final String? deletedAt;
}
